packages/pages/src/dev/server/ssr/getLocalData.ts

```diff
@@ -89,8 +84,31 @@ export const getLocalDataForEntity = async (
       throw err;
     }
   }
+};
+
+export const getLocalDataForEntity = async (
+  entityId: string,
+  locale: string
+) => {
+  const localData = getLocalData((data) => {
+    return data.id?.toString() === entityId && data.locale === locale;
+  });
+  if (!localData) {
+    throw new Error(
+      `No localData files match entityId and locale: ${entityId} ${locale}`
+    );
+  }
+  return localData;
+};
 
-  throw new Error(
-    `No localData files match entityId and locale: ${entityId} ${locale}`
-  );
+export const getLocalDataForSlug = async (slug: string, locale: string) => {
```

Comment:
if there's more than one localData file generated for this slug field (coming from different streams), we should throw an error or inform user of that. I think currently this would just pick whatever file it gets to first?

Reply:
I'm not sure exactly where this kind of validation should live but I don't think it should hurt to add it here for now!